File: server/routes/deleteNote.js
import { Router } from "express";
import { verifyToken } from "../middleware/auth.js";
import Note from "../models/notes.js";
const router = Router();

router.get("/deleteNote", verifyToken, async (req, res) => {
  console.log("deleting note", req.query, req.user.user_id);
  try {
    await Note.deleteOne({
      note_id: req.query.note_id,
      user_id: req.user.user_id,
    });
    res.status(200).send({ message: "Note deleted successfully" });
  } catch (err) {
    console.error(err);
  }
});
export default router;

File: server/deleteNote.go
package main

import (
	"encoding/json"
	"fmt"
	"net/http"
)

type NoteId struct {
	Note_id string `json:"note_id"`
}

func (db *Database) deleteNote(w http.ResponseWriter, r *http.Request) {
	var note_id NoteId
	user_id := r.Context().Value("id")
	json.NewDecoder(r.Body).Decode(&note_id)

	_, err := db.db.Exec("DELETE FROM Notes WHERE userId = ? AND noteId = ?", user_id, note_id.Note_id)

	if err != nil {
		fmt.Println(err)
		http.Error(w, "There has been an error deleting a note", http.StatusInternalServerError)
		return
	}

	fmt.Fprint(w, "Note successfully deleted")
}
